src/VisualStudio/Core/Impl/SolutionExplorer/DiagnosticItem/BaseDiagnosticAndGeneratorItemSource.cs

```diff
@@ -7,187 +7,209 @@
 using System.Collections.Immutable;
 using System.Diagnostics.CodeAnalysis;
 using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Diagnostics;
-using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Elfie.Model;
+using Microsoft.CodeAnalysis.Remote;
 using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.CodeAnalysis.SourceGeneration;
 using Microsoft.VisualStudio.Language.Intellisense;
 using Microsoft.VisualStudio.Shell;
 using Roslyn.Utilities;
 
-namespace Microsoft.VisualStudio.LanguageServices.Implementation.SolutionExplorer
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.SolutionExplorer;
+
+internal abstract partial class BaseDiagnosticAndGeneratorItemSource : IAttachedCollectionSource
 {
-    internal abstract partial class BaseDiagnosticAndGeneratorItemSource : IAttachedCollectionSource
+    private static readonly DiagnosticDescriptorComparer s_comparer = new();
+
+    private readonly IDiagnosticAnalyzerService _diagnosticAnalyzerService;
+    private readonly IAsynchronousOperationListener _listener;
+    private readonly BulkObservableCollection<BaseItem> _items = new();
```

Comment:
note: this became readonly.  instead of creating a *new* collection that overwrites this, we just bulk-update it in place in the batch-queue callback.  The IAttachedCollectionSource is already documented to support that properly:

>     // Remarks:
>    //     If this collection does not change after the initial collection is returned,
>    //     you may return any implementation of IEnumerable. However, if this collection
>    //     does change over time, you may return any collection which implements INotifyCollectionChanged
>    //     (such as an ObservableCollection). If you return a collection implementing INotifyCollectionChanged,
>    //     the collection's consumer will observe changes to the collection and update the
>    //     view on the collection when the collection is modified.